doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -192,6 +198,48 @@ analysis on inductive or coinductive objects (see :ref:`variants`).
      Makes the tactic equivalent to
      :tacn:`induction` :n:`{+, @induction_clause } @induction_principle`.
 
+   .. example:: Using :tacn:`destruct` on the conclusion
+
+      Creates a subgoal for each constructor, substituting the constructor
+      into the conclusion.
+
+      .. rocqtop:: reset none
+
+         Goal forall m n: nat, m + n = n + m.
+
+      .. rocqtop:: out
+
+         intros.
+
+      .. rocqtop:: all
+
+         destruct n.   (* n is an inductive *)
+
+   .. _example_split_hypothesis:
+
+   .. example:: Using :tacn:`destruct` on a hypothesis
+
+      This gives the effect of a :tacn:`split` or :tacn:`constructor` on the
+      hypothesis.  Creates hypotheses for each constructor of the head constant.
+
+      .. rocqtop:: reset none
+
+         Goal forall A B: Prop, A /\ B -> True.
+
+      .. rocqtop:: out
+
+         intros.
+
+      .. rocqtop:: all
+
+         destruct H.   (* H is a hypothesis *)
```

Comment:
There is no practical difference between "destruct on the conclusion" and "destruct on a hypothesis" since the former is really impossible, you need to destruct a hypothesis or a term constructed from hypotheses.
Here, the difference is really between using destruct on a dependent premise (example 1) or a non-dependent premise (example 2).
As said above, I can't see any way in which this resembles `constructor`, and it can resemble "a split" (not `split`) if the hypothesis is a conjunction, but it's way more general.